skopt/utils.py

```diff
@@ -0,0 +1,103 @@
+from collections import Callable
+from time import time
+
+import numpy as np
+from scipy.optimize import OptimizeResult
+
+
+def create_result(Xi, yi, space=None, rng=None, specs=None, models=None):
+    """
+    Return an instance of OptimizeResult() with the required
+    information.
+
+    Parameters
+    ----------
+    * `Xi` [list of lists, shape=(n_iters, n_features)]:
+        Location of the minimum at every iteration.
+
+    * `yi` [array-like, shape=(n_iters,)]:
+        Minimum value obtained at every iteration.
+
+    * `space` [Space instance, optional]:
+        Search space.
+
+    * `rng` [RandomState instance, optional]:
+        State of the random state.
+
+    * `specs` [dict, optional]:
+        Call specifications.
+
+    * `models` [list, optional]:
+        List of fit surrogate models.
+
+    Returns
+    -------
+    * `res` [`OptimizeResult`, scipy object]:
+        OptimizeResult instance with the required information.
+    """
+    res = OptimizeResult()
+    yi = np.asarray(yi)
+    best = np.argmin(yi)
+    res.x = Xi[best]
+    res.fun = yi[best]
+    res.func_vals = yi
+    res.x_iters = Xi
+    res.models = models
+    res.space = space
+    res.random_state = rng
+    res.specs = specs
+    return res
+
+
+def check_callback(callback):
```

Comment:
Maybe a module for collecting all `check_XXX()` functions?. Hopefully in the future we will have more of these `check_blah()` functions to factor out stuff that is done in all the minimisers.

For the moment I'd be ok with keeping it in `utils`.